src/lib/gadgets/bitwise.unit-test.ts

```diff
@@ -0,0 +1,54 @@
+import { ZkProgram } from '../proof_system.js';
+import {
+  Spec,
+  equivalent,
+  equivalentAsync,
+  field,
+  fieldWithRng,
+} from '../testing/equivalent.js';
+import { Fp, mod } from '../../bindings/crypto/finite_field.js';
+import { Field } from '../field.js';
+import { Gadgets } from './gadgets.js';
+import { Random } from '../testing/property.js';
+
+let Bitwise = ZkProgram({
+  publicOutput: Field,
+  methods: {
+    xor: {
+      privateInputs: [Field, Field],
+      method(a: Field, b: Field) {
+        return Gadgets.xor(a, b, 64);
+      },
+    },
+  },
+});
+
+await Bitwise.compile();
+
+let uint = (length: number) => fieldWithRng(Random.biguint(length));
+
+[2, 4, 8, 16, 32, 64, 128].forEach((length) => {
+  equivalent({ from: [uint(length), uint(length)], to: field })(
+    Fp.xor,
+    (x, y) => Gadgets.xor(x, y, length)
+  );
+});
+
+let maybeUint64: Spec<bigint, Field> = {
+  ...field,
+  rng: Random.map(Random.oneOf(Random.uint64, Random.uint64.invalid), (x) =>
+    mod(x, Field.ORDER)
+  ),
+};
+
+// do a couple of proofs
+equivalentAsync({ from: [maybeUint64, maybeUint64], to: field }, { runs: 3 })(
```

Comment:
this test is missing an `await` and I tried it without -- it doesn't actually succeed

I made the same error in the range check unit test (but that one seems to work) -- shoutout to @MartinMinkov for spotting this mistake in his PR!

The test fails because
1) you use numbers for `2**64` which is not exact; use bigints
2) because you test for `x >`, `y >` but it should be `>=`

Reply:
@Trivo25 can you fix the range check unit test as well?

Reply:
yep!